src/util/http_utils.js

```diff
@@ -597,8 +597,15 @@ function check_headers(req, options) {
     if (isNaN(req_time) && !req.query.Expires && is_not_anonymous_req) {
         throw new options.ErrorClass(options.error_access_denied);
     }
-
-    if (Math.abs(Date.now() - req_time) > config.AMZ_DATE_MAX_TIME_SKEW_MILLIS) {
+    // future request should throw AccessDenied with request is no valid yet message
+    // we add a grace period of one second
+    if (req_time > Date.now() - 1000) {
```

Comment:
```suggestion
    if (req_time > Date.now() - 2000) {
```
One second sounds a bit unforgiving

Reply:
sure but notice that it should + 2000 and not  -2000